text/0314-cdk-construct-hub.md

```diff
@@ -0,0 +1,626 @@
+for, in the programming language they want to use.
+
+Construct Hub automatically indexes all construct libraries published to the
+various general purpose package managers, so that developers have a single place
+to search for constructs, regardless of preferred programming language or
+provisioning engine. Developers will find constructs published by AWS, partners,
+and individual developers. Constuct Hub also generates unified documentation for
+the constructs in all supported programming languages.
+
+Construct Hub is also available as a deploymable CDK construct library, so
+that enterprises and users with specific needs can host a private instance of
+the catalog, making interally developed constructs easier for teams to find.
+Just like the public instance of Construct Hub scans public repositories for
+construct libraries, the private Construct Hub instance can be configured to
+scan private repositories like AWS CodeArtifact, JFrog Artifactory, and others.
+
+### Why should I use this feature?
+
+The Construct Hub is a centralized place that helps discover and share
+custom-made constructs compatible with the AWS Cloud Development Kit (CDK), CDK
+for Kubernetes (CDK8s), CDK for Terraform (CDKtf), and other constructs domains
+such as Projen. It shortens the time it takes to ship cloud applications by
+allowing developers to discover constructs that address their infrastructure
+requirements, and provides a unified cross-language documentation experience to
+help developers integrate those constructs into their application.
+
+Customers can also deploy a private instance of Construct Hub to facilitate
+sharing infrastructure abstractions published to a private package registry
+within the organization.
+
+### Where can I find the Construct Hub?
+
+The public instance of Construct Hub is at: `${TBD}`.
+
+### How do I publish a package to Construct Hub?
+
+Packages published to the public npm registry (npmjs.com) are automatically
+indexed by the public instance of Construct Hub as long as they meet the
+following prerequisites:
+
+1. Cross-language support has been configured correctly using [jsii]
+1. The `cdk` keyword is referenced (in the `keywords` key of `package.json`)
+
+[jsii]: https://aws.github.io/jsii
+
+### Can I opt packages out of the public instance of Construct Hub?
+
+There is currently no self-service way to un-list a package from the Construct
+Hub. If you would like your package to be removed from the public instance of
+Construct Hub, please file an issue in GitHub. You will be required to prove
+ownership of the package(s) you are requesting un-listing for.
+
+### Can I trust all constructs listed on Construct Hub?
+
+Construct Hub does not perform a comprehensive audit of packages it indexes.
+You should not trust packages found via the Construct Hub any differently
+from packages found via other package search tools or registries, such as
+npmjs.com, Maven Central, PyPI, NuGet, etc.
+
+## Specification
```

Comment:
@RomainMuller, I think we need to expand this section with an exact list of features we plan to deliver for each page (and also ideally some form of a roadmap for future versions). @addihorowitz has most of this documented internally, but it should be part of this document.

Reply:
The reason it's important is that this will help us make sure the design actually covers all the required features